archive/schema.go

```diff
@@ -221,16 +252,13 @@ type OpenOptions struct {
 	DataSource iosrc.Source
 }
 
-func OpenArchive(root string, oo *OpenOptions) (*Archive, error) {
-	if root == "" {
-		return nil, errors.New("no archive directory specified")
-	}
-	m, mtime, err := MetadataRead(filepath.Join(root, metadataFilename))
+func OpenArchive(root iosrc.URI, oo *OpenOptions) (*Archive, error) {
```

Comment:
This change necessitates a call to iosrc.ParseURI and attendant error handling before almost every call to OpenArchive. Can we move that inside this function and keep its signature unchanged?

Reply:
idk, @nwt are you ok in this case for `CreateOrOpenArchive` to have to do `OpenArchive(root.String())`? Doesn't seem like a big deal, computers are fast, but there is unsavoriness on both branches of the decision tree.

Reply:
Then how about this?
```go
func CreateOrOpenArchive(root string, co *CreateOptions, oo *OpenOptions) (*Archive, error) {
	rooturi, err := iosrc.ParseURI(root)
	if err != nil {
		return fmt.Errorf("invalid root: %s, err)
	}
	...
	return openArchive(rooturi, oo)
}

func OpenArchive(root string, oo *OpenOptions) (*Archive, error) {
	rooturi, err := iosrc.ParseURI(root)
	if err != nil {
		return fmt.Errorf("invalid root: %s", err)
	}
	return openArchive(rooturi, oo)
}

func openArchive(root iosrc.URI, oo *OpenOptions) (*Archive, error) {
	...
}
```